moveit_planners/chomp/chomp_motion_planner/include/chomp_motion_planner/chomp_optimizer.h

```diff
@@ -84,25 +84,21 @@ class ChompOptimizer
   inline double getPotential(double field_distance, double radius, double clearence)
   {
     double d = field_distance - radius;
-    double potential = 0.0;
 
-    // three cases below:
-    if (d >= clearence)
+    if (d >= clearence)  // everything is fine
     {
-      potential = 0.0;
+      return 0.0;
     }
-    else if (d >= 0.0)
+    else if (d >= 0.0)  // transition phase, no collision yet
     {
       double diff = (d - clearence);
-      double gradient_magnitude = diff * clearence;  // (diff / clearance)
-      potential = 0.5 * gradient_magnitude * diff;
+      double gradient_magnitude = diff / clearence;
+      return 0.5 * gradient_magnitude * diff;  // 0.5 * (d - clearance)^2 / clearance
```

Comment:
It may be worth referencing the formula here, I don't remember by heart if the CHOMP papers have it.

Reply:
Which formula do you want to reference? I already added the formula as a comment.

Reply:
I saw the comment. I meant if there's a good reference an interested reader could go to.

Reply:
Ah. Ok. No idea, where to find such a reference.
I just verified the formula using my mathematical common sense :wink: